src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,100 @@
+package duke;
+
+import java.time.LocalDate;
+
+import duke.command.AddDeadline;
+import duke.command.AddEvent;
+import duke.command.AddToDo;
+import duke.command.ByeCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.FindCommand;
+import duke.command.HelpCommand;
+import duke.command.ListCommand;
+
+/**
+ * Represents a parser that handles input from a user.
+ */
+public class Parser {
+
+    /**
+     * Parses the input provided and returns a command.
+     * @param input Input string to be parsed.
+     * @return A command after parsing the input.
+     * @throws DukeException If invalid command is given.
+     */
+    public static Command parse(String input) throws DukeException {
+
+        String[] processedInput = input.split(" ");
+        String command = processedInput[0];
+        String description = processDescription(processedInput);
+
+        switch (command) {
+        case "help":
+            return new HelpCommand();
+        case "list":
+            return new ListCommand();
+        case "bye":
+            return new ByeCommand();
+        case "done":
+            if (processedInput.length == 1) {
+                throw new DukeException("Please enter a duke.task number to mark done");
+            }
+            return new DoneCommand(Integer.parseInt(processedInput[1]));
+        case "todo":
+            return new AddToDo(command, description);
+        case "event":
+            try {
+                return processEvent(input);
+            } catch (Exception e) {
+                e.printStackTrace();
+                throw new DukeException("\nPlease enter a valid format '/at YYYY-MM-DD XXXX-YYYY'");
+            }
+        case "deadline":
+            try {
+                return processDeadline(input);
+            } catch (Exception e) {
+                e.printStackTrace();
+                throw new DukeException("\nPlease enter a valid format '/by YYYY-MM-DD TIME'");
+            }
+        case "delete":
+            if (processedInput.length == 1) {
+                throw new DukeException("Please enter a duke.task number to delete");
+            }
+            return new DeleteCommand(Integer.parseInt(processedInput[1]));
+        case "find":
+            return new FindCommand(processedInput[1]);
+        default:
+            throw new DukeException("Invalid duke.command. Please enter a valid one");
+        }
+    }
+
+    private static AddDeadline processDeadline(String input) throws Exception {
```

Comment:
Should the class name be Deadline instead of AddDeadline? The latter sounds more like what we would usually name a method.